### PR TITLE
Update DeveloperGuide with UML diagrams and proposed flag feature

### PR DESCRIPTION
**Pull request by ryantanlien**
- Updates the remaining UML diagrams that have not yet been updated by Ryan C's PR
-  Added the text section for proposed flag feature (Will elaborate more with UML diagrams in a later PR)

#95

### CODE REVIEW

File: docs/diagrams/BetterModelClassDiagram.puml [hunk 1]
```diff
@@ -4,18 +4,14 @@ skinparam arrowThickness 1.1
 skinparam arrowColor MODEL_COLOR
 skinparam classBackgroundColor MODEL_COLOR
 
-AddressBook *-right-> "1" UniquePersonList
-AddressBook *-right-> "1" UniqueTagList
-UniqueTagList -[hidden]down- UniquePersonList
-UniqueTagList -[hidden]down- UniquePersonList
+AddressBook *-right-> "1" UniqueApplicantList
```
Comment (RyanCheungJF) [inline docs/diagrams/BetterModelClassDiagram.puml:7]:
```suggestion
Linkedout *-right-> "1" UniqueApplicantList
```

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -234,9 +234,22 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+to have a custom ordering for the flagging feature.
+
+As such, the flag feature alters the `UniqueApplicantList` by changing its internal implementation from an
+`ObservableArrayList` to an `ObservablePriorityQueue`. Since an `ObservablePriorityQueue` does not exist in
+the Java library, the flag feature comes with the team's own design for an `ObservablePriorityQueue`.
+
+The `ObservablePriorityQueue` implements the Java Collections, Iterable, PriorityQueue and Observable interfaces,
+and exposes all related functionality from these relevant interfaces. 
+
+`Applicants` will also be edited to contain a boolean `flagged` for use as a comparator in the `ObservablePriorityQueue`
```
Comment (RyanCheungJF) [inline docs/DeveloperGuide.md:251]:
```suggestion
`Applicant` will also be edited to contain a boolean `flagged` for use as a comparator in the `ObservablePriorityQueue`
```

**Review by RyanCheungJF — COMMENTED** [review]
Small changes, but other than that LGTM!

**Review by RyanCheungJF — CHANGES_REQUESTED** [review]
My apologies, but need to generate the pictures and update them as well.

**Comment by ryantanlien** [comment]
Done with sorting the merge conflicts!

**Review by kaiyichen — APPROVED** [review]
LGTM!!

**Review by RyanCheungJF — APPROVED** [review]

**Pull request merged**